src/chat/narrowsSelectors.js

```diff
@@ -164,7 +144,35 @@ export const getShownMessagesForNarrow: Selector<$ReadOnlyArray<Message | Outbox
         return messagesForNarrow;
       }
 
-      return messagesForNarrow.filter(item => !shouldBeMuted(item, narrow, subscriptions, mute));
+      // TODO: What about starred messages, and the starred-messages view?
+      //   Do we really want to filter starred messages out because the
+      //   stream or topic is muted?
```

Comment:
Nit:

```diff
      // TODO: What about starred messages, and the starred-messages view?
      //   Do we really want to filter starred messages out because the
-     //   stream or topic is muted?
+     //   topic is muted?
```

It looks like the code at this point is just filtering out starred messages when the topic is muted, without checking the stream.

(I see this is made irrelevant later in the series, with the fix for the starred narrow, though.)

Reply:
Ah, yeah.

Filtering on account of stream *or* topic would be the thing that makes any sense, if we were going to filter for either of them here. That way it would be parallel to what we do in the all-messages narrow.

But this change was written to be NFC, so it reflects what the old code was doing before this branch. I guess it turns out that meant filtering the starred-messages narrow for muted topics only, but I didn't notice that when writing this comment and instead imagined it was doing the arguably-reasonable thing.

Reply:
Changed it to say "because the topic (or stream) is muted". In the context of asking this question, the stream case is still relevant -- the question is about hypothetical behaviors we could choose, and if the answer were "yes" it'd probably be "yes" to both. But hopefully that's less confusing in conjunction with actual behavior that's just about the topics.